File: string_1/FirstTwo.java
/**
 * Given a string, return the string made of its first two chars, so the
 * String "Hello" yields "He". If the string is shorter than length 2, return
 * whatever there is, so "X" yields "X", and the empty string "" yields the
 * empty string "". Note that str.length() returns the length of a string.
 * <ul>
 *  <li>firstTwo("Hello") → "He"
 *  <li>firstTwo("abcdefg") → "ab"
 *  <li>firstTwo("ab") → "ab"
 * </ul>
 * 
 * @author LZ-FSDev
 * @see https://codingbat.com/prob/p163411
 * @since 17.0.1
 * @version 0.0.1
 */
public class FirstTwo {
    public static void main(String[] args) {
        System.out.println("firstTwo(\"Hello\") -> " + firstTwo("Hello"));
        System.out.println("firstTwo(\"abcdefg\") -> " + firstTwo("abcdefg"));
        System.out.println("firstTwo(\"ab\") -> " + firstTwo("ab"));
    }

    /**
     * Given a string, return the string made of its first two chars, so the
     * String "Hello" yields "He". If the string is shorter than length 2, 
     * return whatever there is, so "X" yields "X", and the empty string "" 
     * yields the empty string "". Note that str.length() returns the length
     * of a string.
     * 
     * @param str Used to grab the first two characters or less if exists.
     * @return The first two characters of str, or whatever is part of str
     * if its length is less than 2.
     * @since 0.0.1
     */
    public static String firstTwo(String str) {
        if (str.length() < 2) {
            return str;
        }
        return str.substring(0,2);
    }
}

File: string_1/FirstTwo.py
'''
Given a string, return the string made of its first two chars, so the String
"Hello" yields "He". If the string is shorter than length 2, return whatever
there is, so "X" yields "X", and the empty string "" yields the empty string
"".
first_two('Hello') → 'He'
first_two('abcdefg') → 'ab'
first_two('ab') → 'ab'
'''

def first_two(str):
    if len(str) <= 2:
        return str
    return str[0:2]

print("first_two('Hello') -> " + str(first_two('Hello')))
print("first_two('abcdefg') -> " + str(first_two('abcdefg')))
print("first_two('ab') -> " + str(first_two('ab')))
print("first_two(' o') -> " + str(first_two(' o')))
print("first_two('') -> " + str(first_two('')))
print("first_two('x') -> " + str(first_two('x')))
